### ISSUE TITLE
The resulting video is very blurred

### ISSUE THREAD

**Issue by wubangjunjava** [neuralchen/SimSwap#89]
The original video is very clear, and the video generated after face change is very blurred. What is the reason for this? How to solve this problem, I want to get HD video. Thank you.

I will wait for your reply online

**Comment by woctezuma** [comment]
What about the original photo?

**Comment by wubangjunjava** [comment]
[Original video 原始视频](https://artsignimg.trsoft.xin/artsign/temp/Original_video.mp4)
[Original image 原始图片](https://artsignimg.trsoft.xin/artsign/temp/input.png)
[output video 输出结果视频](https://artsignimg.trsoft.xin/artsign/temp/output.mp4)

It can be found that the output video is much more blurred than the original video。对比可以发现，输出后的视频比原始视频模糊很多

**Comment by wubangjunjava** [comment]
> What about the original photo?

Original video 原始视频
Original image 原始图片
output video 输出结果视频

It can be found that the output video is much more blurred than the original video。对比可以发现，输出后的视频比原始视频模糊很多

**Comment by NNNNAI** [comment]
Cause the resolution of the swapped results is 224*224. Simswap is not designed for high resolution face swapping.

**Comment by wubangjunjava** [comment]
> Cause the resolution of the swapped results is 224*224. Simswap is not designed for high resolution face swapping.


能否在合成视频之前，将临时图片进行一次超清处理。
Whether the temporary pictures can be super cleared once before synthesizing the video.

**Comment by NNNNAI** [comment]
You can refer to the [colab](https://colab.research.google.com/github/woctezuma/SimSwap-colab/blob/gpen/SimSwap_images.ipynb) which is created by @woctezuma .

**Comment by woctezuma** [comment]
Yes, one could use a super-resolution tool on the temporary pictures. I like GPEN, but keep in mind that it only improves faces.

**Comment by wubangjunjava** [comment]
> Yes, one could use a super-resolution tool on the temporary pictures. I like GPEN, but keep in mind that it only improves faces.

Thank you very much for your answers, which is very important to me. I'll try GPEN tomorrow. Thank you very much. It's a good open source project

**Comment by wubangjunjava** [comment]
> Yes, one could use a super-resolution tool on the temporary pictures. I like [GPEN](https://github.com/yangxy/GPEN), but keep in mind that it only improves faces.

I use GPEN to optimize it. The video generated 405 pictures in total. It took 11 minutes to complete GPEN processing. My graphics card is a quarter of T4 graphics card. How can I optimize the processing speed? Looking forward to your reply?

我使用了GPEN 来优化。视频一共生成了405张图片，GPEN 处理完成耗时 11分钟，我的显卡是 1/4 的 T4 显卡。我该如何优化处理速度？非常期待您的回复?

**Comment by woctezuma** [comment]
See https://github.com/yangxy/GPEN/issues as I cannot help you optimize GPEN.

**Comment by wubangjunjava** [comment]
> Please see at https://github.com/yangxy/GPEN/issues as I cannot help you optimize GPEN.

Although it didn't solve my problem, thank you very much for bringing such a good project. thank you

**Comment by wubangjunjava** [comment]
非常抱歉再次打扰，我的显卡使用率只有27%左右，怎么样才能充分使用显卡的性能呢？这对我非常重要，期待您的回复。谢谢

I'm sorry to bother you again. My graphics card utilization rate is only about 27%. How can I make full use of the performance of my graphics card? This is very important to me and I look forward to your reply. thank you！

![20210815120252](https://user-images.githubusercontent.com/5328798/129466860-b13bf8a7-7848-44a3-861d-643ecf3e4f0f.jpg)
![20210815120234](https://user-images.githubusercontent.com/5328798/129466867-3de5319f-da66-4bb5-97f5-9cc48b451486.jpg)

**Comment by wubangjunjava** [comment]
> Please see at https://github.com/yangxy/GPEN/issues as I cannot help you optimize GPEN.

非常抱歉再次打扰，我的显卡使用率只有27%左右，怎么样才能充分使用显卡的性能呢？这对我非常重要，期待您的回复。谢谢

I'm sorry to bother you again. My graphics card utilization rate is only about 27%. How can I make full use of the performance of my graphics card? This is very important to me and I look forward to your reply. thank you！

![20210815120234](https://user-images.githubusercontent.com/5328798/129466916-52e0eb8c-88ce-4096-a413-da0a3d369591.jpg)
![20210815120252](https://user-images.githubusercontent.com/5328798/129466920-308de71c-f9ee-421b-8818-5a38348cbdd0.jpg)

**Comment by rayscodelife** [comment]
@wubangjunjava 我也遇到相同的问题，GPU使用率过低，您后来找到方案了吗
